Crash on stream

Got the following crash when I try to stream today:

```
{"level":"info","ts":"2023-09-26T17:58:44Z","caller":"airplay/airplay.go:111","msg":"airplay is now running.  Press CTRL-C to exit."}
{"level":"info","ts":"2023-09-26T17:58:44Z","logger":"interactionHandler","caller":"discord/interactions.go:81","msg":"connected to guild","guildID":"80555577431822336"}
panic: runtime error: makeslice: len out of range

goroutine 54 [running]:
github.com/Trojan295/discord-airplay/pkg/codec.StreamDCAData({0x84f2d8, 0xc00050a440}, {0x84a320, 0xc00011f560}, 0xc0004df4a0, 0xc0004be040)
	/src/pkg/codec/opus.go:34 +0x152
github.com/Trojan295/discord-airplay/pkg/discord.(*DiscordVoiceChatSession).SendAudio(0xc000030680, {0x84f2d8, 0xc00050a440}, {0x84a320, 0xc00011f560}, 0x0?)
	/src/pkg/discord/voicechat.go:61 +0xba
github.com/Trojan295/discord-airplay/pkg/bot.(*GuildPlayer).playPlaylist(0xc0001e8300, {0x84f2d8, 0xc000114440})
	/src/pkg/bot/player.go:329 +0x996
github.com/Trojan295/discord-airplay/pkg/bot.(*GuildPlayer).Run(0xc0001e8300, {0x84f2d8, 0xc000114440})
	/src/pkg/bot/player.go:267 +0x691
github.com/Trojan295/discord-airplay/pkg/discord.(*InteractionHandler).GuildCreate.func1()
	/src/pkg/discord/interactions.go:84 +0x3f
created by github.com/Trojan295/discord-airplay/pkg/discord.(*InteractionHandler).GuildCreate
	/src/pkg/discord/interactions.go:83 +0x1db
```

Any ideas?

--- a/cmd/airplay-local/main.go
+++ b/cmd/airplay-local/main.go
@@ -1,0 +1,46 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Trojan295/discord-airplay/pkg/codec"
+	"github.com/Trojan295/discord-airplay/pkg/sources"
+	"go.uber.org/zap"
+)
+
+var (
+	logger         *zap.Logger
+	ctx            context.Context
+	youtubeFetcher *sources.YoutubeFetcher
+)
+
+func main() {
+	logger, _ = zap.NewProduction()
+	defer logger.Sync()
+
+	ctx = context.Background()
+	youtubeFetcher = sources.NewYoutubeFetcher()
+
+	songs, err := youtubeFetcher.LookupSongs(ctx, "https://www.youtube.com/watch?v=AjiSzMFEDgo&pp=ygUNZGVhZG1hdTUgbGl2ZQ%3D%3D")
+	//songs, err := youtubeFetcher.LookupSongs(ctx, "https://www.youtube.com/watch?v=AjiSzMFEDgo")
+	if err != nil {
+		logger.Fatal("failed to get songs", zap.Error(err))
+	}
+
+	logger.Info("got songs", zap.Any("songs", songs))
+
+	reader, err := youtubeFetcher.GetDCAData(ctx, songs[0])
+	if err != nil {
+		logger.Fatal("failed to get dca data", zap.Error(err))
+	}
+
+	opusChan := make(chan []byte, 10)
+	go func() {
+		codec.StreamDCAData(ctx, reader, opusChan, nil)
+		close(opusChan)
+	}()
+	for data := range opusChan {
+		fmt.Println(data)
+	}
+}

--- a/pkg/sources/youtube.go
+++ b/pkg/sources/youtube.go
@@ -74,19 +74,23 @@ func (s *YoutubeFetcher) GetDCAData(ctx context.Context, song *bot.Song) (io.Rea
 	go func(w io.WriteCloser) {
 		defer w.Close()
 
-		ytArgs := s.getYTdlpGetDataArgs(song)
-		ytCmd := strings.Join(append([]string{"yt-dlp"}, ytArgs...), " ")
+		ytArgs := []string{"-x", "-o", "-", "--force-overwrites", "'" + song.URL + "'"}
 
-		seekArg := ""
+		ffmpegArgs := []string{"-i", "pipe:0"}
 		if song.StartPosition > 0 {
-			seekArg = fmt.Sprintf("-ss %d", int64(song.StartPosition.Seconds()))
+			ffmpegArgs = append(ffmpegArgs, "-ss", song.StartPosition.String())
 		}
-		dcaCmd := exec.CommandContext(ctx, "sh", "-c", fmt.Sprintf("%s | ffmpeg -i pipe: %s -f s16le -ar 48000 -ac 2 pipe:1 | dca", ytCmd, seekArg))
+		ffmpegArgs = append(ffmpegArgs, "-f", "s16le", "-ar", "48000", "-ac", "2", "pipe:1")
+
+		downloadCmd := exec.CommandContext(ctx,
+			"sh", "-c", fmt.Sprintf("yt-dlp %s | ffmpeg %s | dca",
+				strings.Join(ytArgs, " "),
+				strings.Join(ffmpegArgs, " ")))
 
 		bw := bufio.NewWriterSize(writer, downloadBuffer)
-		dcaCmd.Stdout = bw
+		downloadCmd.Stdout = bw
 
-		if err := dcaCmd.Run(); err != nil {
+		if err := downloadCmd.Run(); err != nil {
 			log.Printf("while executing get DCA data pipe: %v", err)
 		}
 
@@ -96,8 +100,4 @@ func (s *YoutubeFetcher) GetDCAData(ctx context.Context, song *bot.Song) (io.Rea
 	}(writer)
 
 	return reader, nil
-}
-
-func (s *YoutubeFetcher) getYTdlpGetDataArgs(song *bot.Song) []string {
-	return []string{"-x", "-o", "-", "--force-overwrites", song.URL}
 }

